### ISSUE TITLE
Duplicate body tag

### ISSUE THREAD

**Issue by issunboshi** [sourcejs/Source#160]
Hi,

I'm using Source to create a living styleguide and I really enjoy using it and how easy it is to use.

I have one problem, I have created a new view to display pages made up of individual components, however when I view a page within a browser it has two body tags. The one I have created inside of the view file and another which wraps any included files within the page. The view was duplicated from the spec.js file in user/core/views/.

Here is the view file:
```
<!DOCTYPE html>
<!--[if IE 9]>         <html lang="en" class="ie9 no_js"> <![endif]-->
<!--[if gt IE 9]><!--> <html lang="en" class="no_js"> <!--<![endif]-->
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=9"/>
    <meta content='width=device-width, initial-scale=1' name=viewport />
    <title><%= info.title %></title>
    <meta name="author" content="<%= info.author %>">
    <% if (info.keywords !== '') { %>
    <meta name="keywords" content="<%= info.keywords %>">
    <% } %>

    <link href="/assets/css/app.css" rel="stylesheet">

</head>
<body>

    <div class="container">
        <%- content %>
    </div>

    <script src="/assets/js/jspm_packages/system.js"></script>
    <script src="/assets/js/config.js"></script>
    <script>
        System.import('app/app');
    </script>
</body>
</html>

```

Here is an example file which uses the view:

```
<%- include('../../specs/header/header.html', {showNav: true, background: '/assets/img/sample_header.jpg', title: 'Lorem', titleModifiers: ''}) %>
<div class="contained_content" role="main">
	<%- include('../../specs/intro/intro.html', {title: 'A title', text: '<a href="/" class="intro__link">Who we are.</a> Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet, consectetur adipisicing elit. Perferendis, consequuntur!.', modifier: 'standout'}) %>
    <%- include('../../specs/intro/intro.html', {title: 'A title', text: '<a href="/" class="intro__link">Who we are.</a> Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia, at!.', modifier: 'standout'}) %>
    <%- include('../../specs/intro/intro.html', {title: 'A title', text: '<a href="/" class="intro__link">Who we are.</a> Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quaerat, quasi..', modifier: 'standout'}) %>
	
</div>

<div class="contained_content contained_content--wide">
    <%- include('../../specs/content_blocks/content_block.html', {items: [{mainTitle: "Engineering career", subTitle: "Find an", link: "http://www.bbc.co.uk", layout: 'normal'}, {mainTitle: "Professional career", subTitle: "Find a", link: "http://www.bbc.co.uk", layout: 'normal'}, {mainTitle: "Graduate scheme", subTitle: "Find a", link: "http://www.bbc.co.uk", layout: 'normal'}]}) %>
</div>

<%- include('../../specs/footer/footer.html') %>

```

None of the included files makes reference to the <body> tag.

Here is the output when the source is viewed:

```

<!DOCTYPE html>
<!--[if IE 9]>         <html lang="en" class="ie9 no_js"> <![endif]-->
<!--[if gt IE 9]><!--> <html lang="en" class="no_js"> <!--<![endif]-->
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=9"/>
    <meta content='width=device-width, initial-scale=1' name=viewport />
    <title>Homepage</title>
    <meta name="author" content="Anonymous">

    <link href="/assets/css/app.css" rel="stylesheet">

</head>
<body>

    <div class="container">
        <body><header class="header" style="background-image: url('/assets/img/sample_header.jpg');" role="banner">
	
    <div class="header__nav contained_content contained_content--wide">

<nav class="nav js-nav" role="navigation">
    <div class="nav__header">
        <svg class="svg-icon nav__header_logo" aria-labelledby="title">
            <title>Homepage</title>
            <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#company_logo--white"></use>
        </svg>
        <a href="#" class="nav__trigger js-nav-trigger" data-nav-target="js-nav__list">Menu</a>
    </div>

    <ul class="nav__list js-nav__list">
        
        <li class="nav__item nav__logo_item">
            <a href="/" class="nav__link nav__logo_link">
                <svg class="svg-icon nav__logo" aria-labelledby="title" width="40px" height="10px">
                    <title>Homepage</title>
                    <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#company_logo--white" width="40px" height="10px"></use>
                </svg>
            </a>
        </li>
        
        
        <li class="nav__item"><a href="http://www.bbc.co.uk" class="nav__link">Who we are</a></li>
        
        <li class="nav__item"><a href="http://www.bbc.co.uk" class="nav__link">We're growing</a></li>
        
        <li class="nav__item"><a href="http://www.bbc.co.uk" class="nav__link">We're different</a></li>
        
        <li class="nav__item"><a href="http://www.bbc.co.uk" class="nav__link">Engineering</a></li>
        
        <li class="nav__item"><a href="http://www.bbc.co.uk" class="nav__link">Professional</a></li>
        
        <li class="nav__item"><a href="http://www.bbc.co.uk" class="nav__link">Graduates</a></li>
        
        
        <li class="nav__item nav__secondary_item"><a href="/" class="nav__link nav__logo_link">Login</a></li>
        
    </ul>
</nav>

    </div>
    
    
    
    <div class="header__content contained_content contained_content--wide">
        <h1 class="header__title  js-flexible-header">company is a global something company.</h1>
    </div>
    
</header>

<div class="contained_content" role="main">
	



    
<section class="intro intro--standout">
	
	<h2 class="intro__title">
		A title
	</h2>
	
    <p class="intro__body">
    	<a href="/" class="intro__link">Who we are.</a> Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet, consectetur adipisicing elit. Perferendis, consequuntur!.
    	
    </p>
</section>

    



    
<section class="intro intro--standout">
	
	<h2 class="intro__title">
		A title
	</h2>
	
    <p class="intro__body">
    	<a href="/" class="intro__link">Who we are.</a> Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia, at!.
    	
    </p>
</section>

    



    
<section class="intro intro--standout">
	
	<h2 class="intro__title">
		A title
	</h2>
	
    <p class="intro__body">
    	<a href="/" class="intro__link">Who we are.</a> Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quaerat, quasi..
    	
    </p>
</section>

	
</div>

<div class="contained_content contained_content--wide">
    

<div class="content_blocks">
    
    <div class="content_blocks__item">
        <a href="http://www.bbc.co.uk" class="content_blocks__link">
        	
            <h2 class="content_blocks__header"> Find an </h2>
            <h3 class="content_blocks__title">Something</h3>
            
        </a>
    </div>
    
    <div class="content_blocks__item">
        <a href="http://www.bbc.co.uk" class="content_blocks__link">
        	
            <h2 class="content_blocks__header"> Find a </h2>
            <h3 class="content_blocks__title">Something</h3>
            
        </a>
    </div>
    
    <div class="content_blocks__item">
        <a href="http://www.bbc.co.uk" class="content_blocks__link">
        	
            <h2 class="content_blocks__header"> Find a </h2>
            <h3 class="content_blocks__title">Something</h3>
            
        </a>
    </div>
    
</div>

</div>



    
<div class="footer contained_content" role="contentinfo">
	<div class="footer__breadcrumb">
		


<nav role="navigation" class="breadcrumb">
    <ul class="breadcrumb__list">
        
        	
        	    <li class="breadcrumb__item"><a href="http://www.bbc.co.uk" class="breadcrumb__link">Overview</a></li>
        	
        	    <li class="breadcrumb__item"><a href="http://www.bbc.co.uk" class="breadcrumb__link">Who we are</a></li>
        	
        

        <li class="breadcrumb__item">Styleguide</li>

    </ul>
</nav>

	</div>
	<div class="footer__equal">
		<p class="footer__content">
			company is an equal opportunities employer.
		</p>
		<div class="footer__social">
			<ul class="social_links">
    <li class="social_links__item">
        <a href="#" class="social_links__link">
            <svg class="svg-icon social_links__icon" aria-labelledby="title" role="img">
                <title>Facebook</title>
                <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#facebook-icon"></use>
            </svg>
        </a>
    </li>

    <li class="social_links__item">
        <a href="#" class="social_links__link">
            <svg class="svg-icon social_links__icon" aria-labelledby="title" role="img">
                <title>Twitter</title>
                <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#twitter-icon"></use>
            </svg>
        </a>
    </li>

    <li class="social_links__item">
        <a href="#" class="social_links__link">
            <svg class="svg-icon social_links__icon" aria-labelledby="title" role="img">
                <title>Youtube</title>
                <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#youtube-icon"></use>
            </svg>
        </a>
    </li>

    <li class="social_links__item">
        <a href="#" class="social_links__link">
            <svg class="svg-icon social_links__icon" aria-labelledby="title" role="img">
                <title>Pinterest</title>
                <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#pinterest-icon"></use>
            </svg>
        </a>
    </li>

    <li class="social_links__item">
        <a href="#" class="social_links__link">
            <svg class="svg-icon social_links__icon" aria-labelledby="title" role="img">
                <title>Linked In</title>
                <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#linkedin-icon"></use>
            </svg>
        </a>
    </li>
</ul>

		</div>
	</div>
	
	<div class="footer__secondary">
		<nav class="footer__nav">
			<ul class="footer__list">
				<li class="footer__list_item">
					<a class="footer__list_link">
						<svg class="svg-icon footer__logo" aria-labelledby="title">
						    <title id="title">company</title>
						    <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#company_logo"></use>
						</svg>
					</a>
				</li>
				<li class="footer__list_item">
					<a href="https://www.company.co.uk" class="footer__list_link">
						Terms &amp; conditions
					</a>
				</li>
				<li class="footer__list_item">
					<a href="https://www.company.co.uk" class="footer__list_link">
						company website
					</a>
				</li>
			</ul>
		</nav>
		<p class="footer__copyright">
			© 2015
		</p>
	</div>
    
</div></body>
    </div>

    <script src="/assets/js/jspm_packages/system.js"></script>
    <script src="/assets/js/config.js"></script>
    <script>
        System.import('app/app');
    </script>
</body>
</html>

```

Is there anything wrong with my view which would cause this to happen? I've tried to find of any similar issues with EJS and didn't come up with anything.

**Comment by operatino** [comment]
Hi,

Thanks for the report. I'm trying to reproduce test case, but seems that problem is somewhere in included files. I can assume that somewhere there's an invalid HTML. Could you please share the zip file with files to include as well?

I could be a thing from JSDom, that is still used in 0.5.3, and will be remove in 0.5.4 or 0.6.0. Could you try to run your example with this SourceJS version? https://github.com/sourcejs/Source/releases/tag/0.5.3-no-jsdom

**Comment by issunboshi** [comment]
Hi @operatino,

Thank you for getting back to me so quickly. I have just updated Source as requested and I only have 1 body tag on my pages, so it's resolved.

Thank you.

**Comment by operatino** [comment]
This is a limited version, without Clarify feature (http://sourcejs.com/specs/example-bootstrap-bundle/carousel/?clarify=true&sections=1). I will merge this fix into 0.5.4, so expect to have it in main release in few weeks.

**Comment by operatino** [comment]
I now merged this fix to `0.5.4-dev`.
